### PR TITLE
Bisect effects refactor

### PR DESCRIPTION
**Pull request by acdlite**
**Not for merge; work-in-progress**

Something in the new fork is causing a topline metrics regression. We're not sure what it is, so we're going to split it into steps and bisect.

We'll use this PR to run syncs to www.

### CODE REVIEW

**Comment by acdlite** [comment]
Oops

File: packages/react-reconciler/src/ReactFiberWorkLoop.new.js [hunk 1]
```diff
@@ -2424,7 +2425,35 @@ function commitMutationEffects(root: FiberRoot, renderPriorityLevel) {
...
+        const deletedChild = nextEffect;
+        const deletions = deletedChild.deletions;
+        if (deletions !== null) {
+          for (let i = 0; i < deletions.length; i++) {
+            const deletion = deletions[i];
+            // Clear the deletion effect so that we don't delete this node more
+            // than once.
+            deletion.flags &= ~Deletion;
+            deletion.deletions = null;
+            commitDeletion(root, deletion, renderPriorityLevel);
```
Comment (bvaughn) [inline packages/react-reconciler/src/ReactFiberWorkLoop.new.js]: If we store a copy of the parent's `deletions` array on each child, and we don't check `deletion.flags & Deletion` in this loop, what stops us from still over-deleting when there is more than one child being deleted?

Comment (acdlite) [reply packages/react-reconciler/src/ReactFiberWorkLoop.new.js]: The Deletion flag gets checked by the outer switch statement. So none of the remaining deletions except the  first one will enter the `case Deletion` block.

Comment (bvaughn) [reply packages/react-reconciler/src/ReactFiberWorkLoop.new.js]: Ah, right. Because the first fiber clears its siblings `Deletion` flag too. Got it.

**Comment by acdlite** [comment]
Oops I did it again

**Comment by bvaughn** [comment]
lol again!

**Comment by acdlite** [comment]
GitHub hates it when I make the head commit match the main branch 🙄

**Pull request closed without merging**